opentelemetry-sdk/src/trace/span_processor.rs

```diff
@@ -729,17 +683,17 @@ mod tests {
 
     #[test]
     fn simple_span_processor_on_end_calls_export() {
-        let (exporter, rx_export, _rx_shutdown) = new_test_exporter();
-        let mut processor = SimpleSpanProcessor::new(Box::new(exporter));
+        let exporter = TestSpanExporter::new();
```

Comment:
I dont know if this approach for testing is any better than the channels based approach earlier. If desired I can use the channels based approach back, but without using cross-bean, but using channels from std itself.

Reply:
Let's build add a buffer in TestSpanExporter to save all spans exported?

Reply:
that'd be using inmemoryexporter then right? This test is merely testing if simpleprocessors calls exporter.export or not.

Reply:
Feels like we can deprecate `TestSpanExporter` in favor of the in memory span exporter?

Reply:
yes that'd be good. let me tackle it as a follow up. There could be other places where we create testexporters, and that can be replaced with InMemoryExporter

Reply:
👍